### PR TITLE
[MDS-6527] Email/notification updates for final applications

### PR DESCRIPTION
**Pull request by taraepp**
## Objective 
- summary of changes uploaded as word doc on ticket

[MDS-6527](https://bcmines.atlassian.net/browse/MDS-6527)

_Why are you making this change? Provide a short explanation and/or screenshots_

### CODE REVIEW

File: services/core-api/app/api/projects/project_summary/models/project_summary.py [hunk 1]
```diff
@@ -929,6 +938,48 @@ def validate_project_summary(cls, data, is_historic):
 
         return errors_found
 
+    def send_status_notification(self, prev_status, mine):
```
Comment (taraepp) [inline services/core-api/app/api/projects/project_summary/models/project_summary.py:941]: just moved this from the resource file, basically, and then added the bit with `noun` and final `elif` if in there

File: services/core-api/app/templates/email/projects/minespace_project_section_email.html [hunk 2]
```diff
@@ -8,7 +8,7 @@
 {{ paragraph(info_block('Mine Name', project.mine_name) +
 info_block('Mine Number', project.mine_no) +
 info_block('Project Title', project.project_title) +
-info_block('Submitted', project.submitted), align='center', brand=brand) }}
+info_block('Submitted', project.submitted), brand=brand) }}
```
Comment (taraepp) [inline services/core-api/app/templates/email/projects/minespace_project_section_email.html:11]: I don't like the centre align. 🤷‍♀️

File: services/core-api/tests/factories.py [hunk 3]
```diff
@@ -938,7 +938,7 @@ class Meta:
 
     class Params:
         minimal = factory.Trait(
-            mine_no=None,
+            mine_no='123456',
```
Comment (taraepp) [inline services/core-api/tests/factories.py:941]: also never like how test emails go out like "Mine No: <empty>" because I'm not sure if it's working properly, so put in a default value here. And made the AmsFinalApplication factory use the correct authorization type below. 😅

File: services/core-api/tests/projects/ams_final_application/resources/test_ams_final_application_resource.py [hunk 4]
```diff
@@ -595,9 +597,21 @@ def test_put_ams_final_app_minespace_edit_toggle(test_client, db_session, auth_h
     assert put_resp.status_code == 200
     assert put_resp_data['editable'] == False
 
+    put_data2 = {
```
Comment (taraepp) [inline services/core-api/tests/projects/ams_final_application/resources/test_ams_final_application_resource.py:600]: I was getting all test emails with the ams application being locked which was throwing me off, so added in this bit to make sure it worked right when it's unlocked.

File: services/core-api/app/api/projects/ams_final_application/models/ams_final_application.py [hunk 5]
```diff
@@ -90,16 +103,213 @@ def update(self,
...
+            reference_table='ams_final_application',
+            reference_email_type=f'ams_app_{event.value.lower()}_email'
+        )
+
+    def _send_resubmit_email(self, core_recipients, minespace_recipients):
+        project_summary = self.project_summary_authorization.project_summary
+        project = project_summary.project
+        mine = project.mine
+        
+        message = f'Updates to EMA major project final application for {project.project_title} has been submitted for {mine.mine_name}'
```
Comment (matbusby-fw) [inline services/core-api/app/api/projects/ams_final_application/models/ams_final_application.py]: Minor grammer thing.  'has' -> 'have' (guessing this text was provided to you)

**Review by matbusby-fw — CHANGES_REQUESTED** [review]
Looks good.  Just noticed one tiny grammar thing

**Review by matbusby-fw — APPROVED** [review]

**Review by asinn134 — APPROVED** [review]
